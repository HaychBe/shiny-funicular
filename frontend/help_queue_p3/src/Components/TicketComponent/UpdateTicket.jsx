import React, { useEffect, useState } from "react";
import axios from "axios";
import {Button, Modal, Form } from "react-bootstrap";

const UpdateTicket = ({
  id,
  oldTitle,
  oldTraineeName,
  oldTrainer,
  oldCohort,
  oldDescription,
  oldTopic,
}) => {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const [title, setTitle] = useState(oldTitle);
  const [traineeName, setTrainee] = useState(oldTraineeName);
  const [trainer, setTrainer] = useState(oldTrainer);
  const [cohort, setCohort] = useState(oldCohort);
  const [description, setDescription] = useState(oldDescription);
  const [topic, setTopic] = useState(oldTopic);
  const [time, setTime] = useState(
    new Date().toLocaleString('en-GB')
  );

const updateData = (e) => {
    e.preventDefault();

    const ticketValues = {
        title: title,
        trainee: traineeName,
        trainer: trainer,
        topic: topic,
        _description: description,
        cohort: cohort,
        dateTime: time,
      };
    
    //   console.log(ticketValues);
      
    setTime(new Date().toLocaleString("en-GB"))
    axios.put("http://localhost:8080/ticket/update/" + id, ticketValues,  {
      headers: {
        'Access-Control-Allow-Origin' : '*'
        }
    })
      .then(function (response) {
        console.log(response);
      })
      .catch(function (error) {
        console.log();
        console.log(error);
      });

      handleClose();
      
  };

  return(

    <>
        <Button 
        variant="info" 
        onClick={handleShow}
        > Update
        </Button>

        <Modal
            show={show}
            onHide={handleClose}
            dialogClassName="ticketModal"
            scrollable={true}
        >
            <Modal.Header closeButton>
                <Modal.Title>Edit Ticket Modal</Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <Form onSubmit={updateData}>
                    <Form.Group>
                        <Form.Label>Trainee Name</Form.Label>
                        <Form.Control
                            type="text"
                            value={traineeName}
                            placeholder="Trainee Name"
                            onChange={(i) => {
                                setTrainee(i.target.value);
                            }}
                        />
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Cohort</Form.Label>
                        <Form.Control
                            as="select"
                            value={cohort}
                            onChange={(i) => {
                                setCohort(i.target.value);
                            }}
                        >
                            <option>CloudNative</option>                        
                            <option>Software Specialist</option>                        
                            <option>DevOps</option>   
                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Topics</Form.Label>
                        <Form.Control
                            as="select"
                            value={topic}
                            onChange={(i) => {
                                setTopic(i.target.value);
                            }}
                        >
                            <option>FrontEnd</option>
                            <option>BackEnd</option>
                            <option>DevOps</option>
                            <option>Cloud Fundamentals</option>

                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Trainer</Form.Label>
                        <Form.Control
                            as="select"
                            value={trainer}
                            onChange={(i)=>{
                                setTrainer(i.target.value);
                            }}
                        >
                            <option>Vinesh Ghela</option>
                            <option>Jordan Harrison</option>
                            <option>Reece Elder</option>
                            <option>Nathan Forester</option>
                            <option>Harry Volker</option>
                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Title</Form.Label>
                        <Form.Control
                            type="text"
                            value={title}
                            onChange={(i) => {
                                setTitle(i.target.value);
                            }}
                        />
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Description</Form.Label>
                        <Form.Control
                            as="textarea"
                            rows={5}
                            value={description}
                            onChange={(i) => {
                                setDescription(i.target.value);
                            }}
                        />
                    </Form.Group>
                </Form>

                <Button
                    onClick = {updateData}
                    type="submit"
                    variant="success"
                    size="lg"
                    block
                >Update Ticket
                
                </Button>
            </Modal.Body>
        </Modal>
    </>
  ) 



}
export default UpdateTicket;