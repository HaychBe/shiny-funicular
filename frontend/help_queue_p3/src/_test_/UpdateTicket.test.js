import React from 'react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import {expect, it} from '@jest/globals';
import UpdateTicket from '../Components/TicketComponent/UpdateTicket';

describe(`Ticket render tests with ticket data props`, () => {

    const ticketValues = {
        title: "title",
        topic: "topic",
        trainee: "traineeName",
        cohort: "cohort",
        trainer: "trainer",
        description: "description",
        dateTime: "time"
    }
    let componentToTest;

beforeEach( () => {
    const TestInstance = create(<UpdateTicket{...ticketValues}/>)
    componentToTest = TestInstance.root;
});

it(`Should render "title" in a title form`, () => {
    const titl = componentToTest.findByProps({ id: "tickTitle"});
    expect(titl.children).toEqual([ticketValues.title]);
});

})