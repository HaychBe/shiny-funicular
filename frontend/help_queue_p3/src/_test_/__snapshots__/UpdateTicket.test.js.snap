// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTicket should match the snapshot every time it is rendered 1`] = `
<button
  className="btn btn-secondary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   Update
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="btn btn-secondary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   Update
</button>
`;
